js/client/modules/@arangodb/testutils/instance-manager.js

```diff
@@ -524,20 +527,48 @@ class instanceManager {
   // / @brief dump the state of the agency to disk. if we still can get one.
   // //////////////////////////////////////////////////////////////////////////////
   dumpAgency() {
+    const dumpdir = fs.join(this.options.testOutputDirectory, `agencydump_${this.instanceCount}`);
+    const zipfn = fs.join(this.options.testOutputDirectory, `agencydump_${this.instanceCount}.zip`);
+    if (fs.isFile(zipfn)) {
+      fs.remove(zipfn);
+    };
+    if (fs.exists(dumpdir)) {
+      fs.list(dumpdir).forEach(file => {
+        const fn = fs.join(dumpdir, file);
+        if (fs.isFile(fn)) {
+          fs.remove(fn);
+        }
+      });
+    } else {
+      fs.makeDirectory(dumpdir);
+    }
     this.arangods.forEach((arangod) => {
       if (arangod.isAgent()) {
         if (!arangod.checkArangoAlive()) {
           print(Date() + " this agent is already dead: " + JSON.stringify(arangod.getStructure()));
         } else {
           print(Date() + " Attempting to dump Agent: " + JSON.stringify(arangod.getStructure()));
-          arangod.dumpAgent( '/_api/agency/config', 'GET', 'agencyConfig');
+          arangod.dumpAgent( '/_api/agency/config', 'GET', 'agencyConfig', dumpdir);
 
-          arangod.dumpAgent('/_api/agency/state', 'GET', 'agencyState');
+          arangod.dumpAgent('/_api/agency/state', 'GET', 'agencyState', dumpdir);
 
-          arangod.dumpAgent('/_api/agency/read', 'POST', 'agencyPlan');
+          arangod.dumpAgent('/_api/agency/read', 'POST', 'agencyPlan', dumpdir);
         }
       }
     });
+    let zipfiles = [];
+    fs.list(dumpdir).forEach(file => {
+      const fn = fs.join(dumpdir, file);
+      if (fs.isFile(fn)) {
+        zipfiles.push(file);
+      }
+    });
```

Comment:
Same question here: Why we add to zip only files but not directories and content of them?

Reply:
`..` is a directory ;)